yt/data_objects/construction_data_containers.py

```diff
@@ -2774,36 +2782,29 @@ def get_data(self, fields=None):
         if fields is None:
             return
 
-        # not sure on the best way to do this
-        if isinstance(fields, list) and len(fields) > 1:
-            for field in fields:
-                self.get_data(field)
-            return
-        elif isinstance(fields, list):
-            fields = fields[0]
+        sph_ptypes = getattr(self.ds, "_sph_ptypes", ["PartType0"])
```

Comment:
I think the reason we don't fall back to `["all"]` is because we want the type being used to have smoothing lengths generated. Maybe that doesn't matter anymore, I'll think about this

Reply:
This assumes a Gadget or Gadget-like dataset--I think we need to have a fallback that checks the particle types in the dataset and selects the one which is likely to have the smoothing length.

Reply:
yeah, that is true! Is there already a place in the code that performs this check? I assume I'm not the first to come across some logic looking for particles with smoothing lengths.

Reply:
There isn't, actually--we always rely on the datasets themselves to specify what the SPH type is. Let me see if I can come up with something.

Which makes me wonder--is there a reason you wouldn't expect `_sph_ptypes` to be defined here?